test/tests/mesh_modifiers/sidesets_between_subdomains/between_vector_blocks.i

```diff
@@ -0,0 +1,27 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 2
+  ny = 2
+  nz = 0
+  zmin = 0
+  zmax = 0
+  elem_type = QUAD4
+[]
+
+[MeshModifiers]
+  [./subdomain_id]
+    type = AssignElementSubdomainID
+    subdomain_ids = '0 0
```

Comment:
I suggest here you give '0 1 2 3', then you can add another mesh modifier trying to add side set between 0 and 3 to make sure no side set will be added.